### PR TITLE
feat(AWS RabbitMQ): Support `virtualHost`

### PR DESCRIPTION
**Pull request by EnterNameHere7**
Closes: #10813

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/rabbitmq.js [hunk 1]
```diff
@@ -45,9 +45,12 @@ class AwsCompileRabbitMQEvents {
         queue: {
           type: 'string',
         },
+        virtualHost: {
+          type: 'string',
+        },
       },
       additionalProperties: false,
-      required: ['basicAuthArn', 'arn', 'queue'],
+      required: ['basicAuthArn', 'arn', 'queue', 'virtualHost'],
```
Comment (jc1arke) [inline lib/plugins/aws/package/compile/events/rabbitmq.js:53]: Maybe the `virtualHost` field should be an optional?

Comment (EnterNameHere7) [reply lib/plugins/aws/package/compile/events/rabbitmq.js]: updated in latest commit

File: lib/plugins/aws/package/compile/events/rabbitmq.js [hunk 2]
```diff
@@ -101,6 +104,21 @@ class AwsCompileRabbitMQEvents {
...
         const dependsOn = this.provider.resolveFunctionIamRoleResourceName(functionObj) || [];
 
+        const sourceAccessConfigurations = [
+          {
+            Type: 'BASIC_AUTH',
+            URI: basicAuthArn,
+          }
+        ]
+
+        // if not specified it will default to /
```
Comment (pgrzesik) [inline lib/plugins/aws/package/compile/events/rabbitmq.js:114]: It's not clear what is the source of that default - is that the default behavior on AWS side? If that's the case, I think we can safely remove this comment from the code

**Review by pgrzesik — COMMENTED** [review]
Thanks @EnterNameHere7 - I have one minor question and we should be good to go - great job 👍

**Comment by pgrzesik** [comment]
@EnterNameHere7 it also looks like there's a check failing in CI - could you please take a look at it?

File: lib/plugins/aws/package/compile/events/rabbitmq.js (hunk 2 above)
Comment (EnterNameHere7) [reply lib/plugins/aws/package/compile/events/rabbitmq.js:114]: it defaults it to a / in AWS will remove the comment

**Comment by jc1arke** [comment]
@pgrzesik any chance on getting this reviewed?

**Comment by pgrzesik** [comment]
Hey @jc1arke - sorry for missing it, the best way to ensure I see the updates is to request a review after pushing changes (just pushing changes doesn't send any notifications).

**Comment by pgrzesik** [comment]
@EnterNameHere7 hey 👋 There seems to be an issue with CI at the moment, could you rebase your changes on top of `main` and push again?

**Comment by EnterNameHere7** [comment]
@pgrzesik I updated the branch with the main branch and also ran prettier and lint to make sure it falls in line, should be good to go

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Sorry, I've missed one thing - documentation. It would be great to add this field to `rabbitmq.md` doc about the event and additionally to `serverless.yml.md` doc as well - could you please handle that @EnterNameHere7? Sorry for not mentioning it sooner

**Comment by EnterNameHere7** [comment]
@pgrzesik sorry that I took so long to respond, documentation updated and updated with main branch

**Review by pgrzesik — APPROVED** [review]
Thank you @EnterNameHere7, great job 👏

**Pull request merged**